### ISSUE TITLE
Should this be private or public

### ISSUE THREAD

**Issue by orangepaydev** [Jacky26/quarkus-interview-practice#1]
https://github.com/Jacky26/quarkus-interview-practice/blob/add1828e78ebd7695d2dbad8db75b245da747673/src/main/java/org/jc/service/DataProcessingService.java#L15

Should this be Private or Public?  Give an explanation as to the benefit and disadvantage.

**Comment by Jacky26** [comment]
In my opinion, to use either Private or Public will be depending on the usage of that method. I made a mistake here, I think this isDuplicate() method should be private in this case as the only usage is within the same class which called by another method. If putting it as public may have potential to be misused by other classes. 

The benefit for declaring private method is providing a clear boundary for the class's functionality, encapsulate it to prevent accessible from outside the class. Hence, we know that this private method will only have usage within the same class, when changing implementation it will not affect the external method outside of the class. 

The disadvantage for a private method basically is the accessibility limitation which mean by not able to use by other class. Hence, we should not create as private method if the functionality is needed by other classes. Besides, there is also challenges to test a private method as it is not accessible from test class. Might need to use other techniques to test them. Normally I will just create some tests on public method and try to cover the scenarios of private method as well.

I think Protected is also suitable access modifier if we want to provides one more level of visibility to the method and also easier for writing test case. Protected gives advantage like accessed within classes in the same package and by subclasses. Assuming we have another class extend to this DataProcessingService class, it provide to use or modify the behavior if necessary

**Comment by orangepaydev** [comment]
good point, so given the above...I notice that some of your method are marked as Public, will you beable to modify them and update them as private in your method.

**Comment by Jacky26** [comment]
Yes, I have updated the `isDuplicate(Customer customer1, Customer customer2)` method to private as we discussed it is used within the same class level only. For the rest of the methods like `convertDataToList(String data)` & `findDuplicateCustomer(List<Customer> customers)` I will remain as public as they are being called in the controller layer.

Do refer to the latest commit
